Fixes for unread counts, device names, and disabling of JS on iDevices

stuartbreckenridge opened the pull request.

stuartbreckenridge committed:
Fixes #3892 - Formatter used for unread counts

stuartbreckenridge committed:
Fixes #4322 - Device names are handled via l8n

Added a `DefaultAccountNames.xcstrings` file that `defaultAccountName` looks to for device names. It has a variation per device. Note that it’s “On My Apple Vision”…no “Pro” modifier.

stuartbreckenridge committed:
Fixes #4323  adds option to disable js on iDevices

brentsimmons reviewed (changes requested):
Looks good! Just a few changes requested.

For the future, let’s separate multiple issues into multiple PRs. (It’s easier and quicker for me to review.)

stuartbreckenridge committed:
Address #4323 Feedback

Note: the diff for Settings.storyboard is large, but it’s only to add a tableviewcell for the enableJS toggle.

stuartbreckenridge committed:
address #4322 feedback

stuartbreckenridge commented on iOS/AppDefaults.swift:
```diff
@@ -189,6 +190,15 @@ final class AppDefaults {
+			AppDefaults.setBool(for: Key.disableJavaScript, newValue)
+		}
+	}
+	
```
Done

brentsimmons commented on iOS/Article/PreloadedWebView.swift:
```diff
@@ -46,6 +48,11 @@ class PreloadedWebView: WKWebView {
 		}
 	}
 	
+	@objc func userDefaultsDidChange(_ sender: Any) {
+		configuration.defaultWebpagePreferences.allowsContentJavaScript = AppDefaults.shared.isArticleContentJavascriptEnabled
+		reload()
+	}
```
This will result in the page being reloaded any time any UserDefaults value changes — which can be surprisingly often. See how the Mac version is handling this, to make sure it reloads only when the JavaScript setting changes.

stuartbreckenridge replied:
Fixed in 402fbd3.

brentsimmons reviewed (changes requested):
One more change requested — how we handle reloading when the setting changes.

stuartbreckenridge committed:
🩹 Amend #4323  JS pref changes cause reload only
